### PR TITLE
CSV Button

### LINKED ISSUE #3: CSV Converter
Add download CSV button with just averages

### PR DESCRIPTION
**Pull request by DavidHZhu**
Fix #3 

Right now it doesn't work but the idea is create a `p5.table()` then add rows to it and pass it to `index.js` where its gonna call `p5.save(table, file_name)`

some1 fix plz I need to shit so bad

### CODE REVIEW

File: src/Game.js [hunk 1]
```diff
@@ -124,6 +125,26 @@ export default class Game {
     return output;
   }
 
+  csv(p5) {
+    const output = {
+      num_generations: this.generations.length,
+      headers: ["Generation","Distance","Speed","Size","Sense"],
+      generations: []
```
Comment (DavidHZhu) [inline src/Game.js]: ah u can probably remove this, was from a copy-paste from above

File: src/Game.js [hunk 2]
```diff
@@ -5,13 +5,13 @@ import {getRandomInt} from "./Helpers/Helpers";
 import Genes from "./Genes";
 
 export default class Game {
-  constructor(width, height, n) {
+  constructor(width, height, n_food) {
```
Comment (devonpmack) [inline src/Game.js]: This is actually the number of creatures and food combined into one variable. It seems appropriate to split it up into two variables instead

Comment (devonpmack) [reply src/Game.js]:
```suggestion
  constructor(width, height, n_food, n_creatures) {
```

File: src/Game.js [hunk 3]
```diff
@@ -5,13 +5,13 @@ import {getRandomInt} from "./Helpers/Helpers";
...
     this.width = width;
     this.height = height;
     this.generations = [];
     this.gen = 0;
 
-    this.randomGen(n);
+    this.randomGen(n_food);
```
Comment (devonpmack) [inline src/Game.js]:
```suggestion
    this.randomGen(n_food, n_creatures);
```

Comment (devonpmack) [reply src/Game.js]: you will also need to update the randomGen function

File: src/Game.js [hunk 4]
```diff
@@ -124,6 +125,29 @@ export default class Game {
     return output;
   }
 
+  csv(p5) {
+    // Table of info
+    let csvRows = new p5.Table();
```
Comment (devonpmack) [inline src/Game.js]:
```suggestion
    let csvTable = new p5.Table();
```

File: src/index.js [hunk 5]
```diff
@@ -22,6 +22,13 @@ let s = (p5) => {
       p5.saveJSON(game.json(), `ns_${game.gen}gens.json`);
     }));
 
+    // To be added
+    buttons.push(new Button(10,100,100,13, "Download CSV", () => {
+      p5.save(game.csv(p5), `ns_${game.gen}gens.csv`);
+      //console.log(game.csv(p5));
+    }));
```
Comment (devonpmack) [inline src/index.js]: Nice work learning how to use my button library 🎉

File: src/Game.js [hunk 6]
```diff
@@ -124,6 +124,28 @@ export default class Game {
...
+    this.generations.forEach((gen) => {
+      let row = csvTable.addRow();
+      row.set("Generation", gen.gen);
+      row.set("Distance", gen.avg_distance);
+      row.set("Speed", gen.avg_speed);
+      row.set("Size", gen.avg_size);
+      row.set("Sense", gen.avg_sense);
+    });
+    
+    return csvTable;
```
Comment (justinpaoli) [inline src/Game.js]: Remove whitespace lines around return

File: src/index.js (hunk 5 above)
Comment (DavidHZhu) [reply src/index.js]: im so good at video games

File: src/Game.js [hunk 7]
```diff
@@ -1,17 +1,18 @@
 import Creature from "./Creature";
 import Food from "./Food";
-import {BLUE, DEAD_BREED, MIN_SIZE, N_FOOD, RED} from "./Constants";
+import {BLUE, DEAD_BREED, MIN_SIZE, N_FOOD, N_CREATURES, RED} from "./Constants";
```
Comment (DavidHZhu) [inline src/Game.js]: bro u imported `N_FOOD` here and u also pass in `n` from the Game constructor on line 8

**Comment by devonpmack** [comment]
I think if we put the constants directly into the code we lose a lot of encapsulation but then again i did that everywhere else so who cares

File: src/Game.js (hunk 1 above)
Comment (devonpmack) [reply src/Game.js]: please clarify

File: src/Game.js (hunk 7 above)
Comment (devonpmack) [reply src/Game.js]: yeah this is fucked

**Pull request merged**